arch/x86/core/x86_mmu.c

```diff
@@ -136,3 +136,110 @@ void _x86_mmu_set_flags(void *ptr, size_t size, u32_t flags, u32_t mask)
 		addr += MMU_PAGE_SIZE;
 	}
 }
+
+#ifdef CONFIG_X86_USERSPACE
+
+/* Helper macros needed to be passed to x86_update_mem_domain_pages */
+#define X86_MEM_DOMAIN_SET_PAGES   (0U)
+#define X86_MEM_DOMAIN_RESET_PAGES (1U)
+/* Pass 1 to reset_pages if reset of mem domain pages is needed else pass a 0*/
+static inline void x86_update_mem_domain_pages(struct k_mem_domain *mem_domain,
+					       u32_t reset_pages)
+{
+	u32_t partition_index;
+	u32_t total_partitions;
+	struct k_mem_partition partition;
+	u32_t partitions_count;
+
+	do {
+		/* If mem_domain doesn't point to a valid location return.*/
+		if (mem_domain == NULL) {
+			break;
+		}
+
+		/* Get the total number of partitions*/
+		total_partitions = mem_domain->num_partitions;
+		__ASSERT(total_partitions <= CONFIG_MAX_DOMAIN_PARTITIONS,
```

Comment:
why is this assertion placed here?
should be in the common mem_domain code, if anywhere